Command line cleanup

nishantjr opened the pull request.

Commit:
test/expected.cpp,spec.rv: Monitor topic "test" => "chatter"

Commit:
test/CppGeneratorTest: Format output string before comparison

Commit:
CppGenerator,test/expected.cpp: monitor node pulishes message only when enabled

nishantjr committed:
INSTALL: instructions for installing rv-monitor, rosmsg

nishantjr committed:
bin/rosmop: Make script thin wrapper around rosmop.Main

nishantjr committed:
Propagate exceptions rather than hiding them

nishantjr committed:
Main: Simplify/generalize argument parsing

nishantjr committed:
Main: tabs to spaces

nishantjr committed:
Main: `-monitorAsNode` => `--monitor-as-node`

nishantjr committed:
Main: Add `--output-prefix` option for specifying output location

msaxena2 commented on src/main/java/rosmop/Main.java:
```diff
@@ -56,99 +56,47 @@
-		}
+	public static void main(String[] _argv)
+	    throws ROSMOPException, LogicException, RVMException, java.io.IOException
+	{
```
For Java most conventions prefer not having the opening brace on a newline.

msaxena2 replied:
Hmm. I guess the outdated tag comes from the fact that I added the comment to an older commit.

nishantjr replied:
I find this extremely hard to read though:

``` 
   private static List<File> expandDirectories(List<String> inputFiles)        
        throws IOException {                                            
        List<File> rvFiles = new ArrayList<File>();                             
        for (String p : inputFiles) {                                           
            File f = new File(p);                                               
            if (f.isDirectory()) {                                              
                rvFiles.addAll(Arrays.asList(f.listFiles()));                   
            }                                                                   
            else {                                                              
                rvFiles.add(f);                                                 
            }                                                                   
        }                                                                       
        return rvFiles;                                                         
    }                                                                           
```

Any more readable suggestions?

msaxena2 replied:
One extra line of indentation before the throw and a newline? It's unreadable as above, I agree, but something along the lines of what's suggested here - https://stackoverflow.com/a/46275233 ?

msaxena2 commented on src/main/java/rosmop/Main.java:
```diff
@@ -56,99 +56,47 @@
+    // Replace each directory in the input args with the list of files it contains
+	private static List<File> expandDirectories(List<String> inputFiles)
+	    throws java.io.IOException
     {
```
Here as well. It's inconsistent in the old code base, but if we're fixing stuff, might as well do it right.

msaxena2 commented on src/main/java/rosmop/Main.java:
```diff
@@ -56,99 +56,47 @@
+            if (f.isDirectory()) {
+                rvFiles.addAll(Arrays.asList(f.listFiles()));
+            }
+            else {
```
Generally, the else is placed right after the closing if brace. Lets stick to it?

msaxena2 commented on src/main/java/rosmop/Main.java:
```diff
@@ -56,99 +56,47 @@
-					pathToFile.lastIndexOf(File.separator)+1) + "rvmonitor";
-			//				System.out.println(pathToOutputNoExt);
-			processMultipleFiles(inputSpecFiles);
+        List<File> rvFiles = new ArrayList<File>();
```
The `<File>` can just be replaced with `<>` on the RHS.

msaxena2 commented on bin/rosmop:
```diff
@@ -1,69 +1,9 @@
 #!/bin/bash
 
```
The original script threw an error if the jar wasn't present, with a suggestion to call `mvn package`. Can we add that check here as well, since it's replacing the original?

nishantjr replied:
The `java` command will already give you an error for that case, no?

msaxena2 replied:
Yeah, it's a class loader exception along with a stacktrace, which in my opinion is much less instructive.

msaxena2 reviewed (commented):
Looks good. My commits (4 earliest ones) need polishing though. We can put your changes onto master, and I can rebase my commits off of those (if you'd like this in immediately), or I'll have the commits ready for your review tomorrow morning.

msaxena2 commented on src/main/java/rosmop/Main.java:
```diff
@@ -36,364 +39,304 @@
+        Options options = new Options();
+        options.addOption("n", "monitor-as-node", false, "Generated monitor can be run as a stand-alone ROS node.");
+        options.addOption(
+            OptionBuilder.withLongOpt("output-prefix")
```
This is deprecated. https://stackoverflow.com/a/35050942 .

nishantjr committed:
Main: Use Option.builder instead of deprecated OptionBuilder

nishantjr committed:
Formatting

nishantjr commented:
@msaxena2 ~Lets drop the previous commits, for now just to do things more modularly~. Addressed all your comments except the braces-on-same-line-issue. I'm ok with having the braces on the same line for the case where the declaration fits in a single 80 character line, and have it on its own line otherwise.

EDIT: Looks like some of my commits modify the code in a way I'd rather not bother redoing. Lets fix the top 4 commits

msaxena2 commented:
@nishantjr Working on cleaning up the first few. Probably a couple of hours.

Commit:
Main,{Cpp,HeaderGen},resources/dl-spec: Gen C++ for dl specs

Commit:
code/{Cpp,Header}Generator: Formatting

msaxena2 commented:
@nishantjr I'm ok with your original choice of brace formatting. For the romsop bash script, I'd rather have the original message instead of java's Class Not Found exception. Otherwise, if you're ok with my commits, we should put these on master.

nishantjr committed:
bin/rosmop: Print user-friendly message when jar doesn't exist

nishantjr committed:
Main: More fixing exception propagation

nishantjr committed:
HeaderGenerator: Don't generate declaration for destructor

A corresponding definition is not defined, causing linker errors.
We got away with this so far because we leak the pointer to the instance created
and never call the destructor.
